<?xml version="1.0"?>

<launch>

  <!-- arguments -->
  <arg name="world_frame" default="/world" />
  <arg name="server" default="192.168.2.230" />
  <arg name="port" default="3883" />

  <arg name="uri_D1" default="radio://0/84/1M/E7E7E7E703" />
  <arg name="frame_D1" default="cf4" />
  <arg name="id_D1" default="D1"/>
  <arg name="v_D" default=".3"/>

  <arg name="uri_D2" default="radio://0/84/1M/E7E7E7E703" />
  <arg name="frame_D2" default="cf4" />
  <arg name="id_D2" default="D2"/>

  <arg name="uri_I" default="radio://0/84/1M/E7E7E7E703" />
  <arg name="frame_I" default="cf3" />
  <arg name="id_I" default="I"/>
  <arg name="v_I" default=".4"/>
  
  <!-- start crazyflie_client -->
  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>
  <!-- run vrpn client -->
  <include file="$(find optitrack_broadcast)/launch/optitrack_broadcast_p300.launch">
    <arg name="server" value="$(arg server)" />
    <arg name="frames" value="$(arg frame_D1) $(arg frame_D2) $(arg frame_DI)" />
  </include>

  <!-- add the intruder -->
  <include file="$(find crazyflie_game)/launch/add_player.launch">
    <arg name="world_frame" default="$(arg world_frame)" />
    <arg name="uri" default="$(arg uri_I)" />
    <arg name="frame" default="$(arg frame_I)" />
    <arg name="id" default="$(arg id_I)"/>
    <arg name="v" default="$(arg v_I)"/>
  </include>

  <!-- add defender 1 -->
  <include file="$(find crazyflie_game)/launch/add_player.launch">
    <arg name="world_frame" default="$(arg world_frame)" />
    <arg name="uri" default="$(arg uri_D1)" />
    <arg name="frame" default="$(arg frame_D1)" />
    <arg name="id" default="$(arg id_D1)"/>
    <arg name="v" default="$(arg v_D)"/>
  </include>

  <!-- add defender 2 -->
  <include file="$(find crazyflie_game)/launch/add_player.launch">
    <arg name="world_frame" default="$(arg world_frame)" />
    <arg name="uri" default="$(arg uri_D2)" />
    <arg name="frame" default="$(arg frame_D2)" />
    <arg name="id" default="$(arg id_D2)"/>
    <arg name="v" default="$(arg v_D)"/>
  </include>

  <!-- game data recorder -->
  <node name="game_recorder" pkg="crazyflie_game" type="game_recorder.py" output="screen">
      <param name="D1" value="$(arg D1)"/>
      <param name="D2" value="$(arg D2)" />
      <param name="I" value="$(arg I)" />
  </node>


</launch>